PowerVCF.psm1

```diff
@@ -5690,6 +5690,348 @@ Export-ModuleMember -Function Get-VCFWsa
+    Get the ID of a VCF identity provider for a specific type.
+    .DESCRIPTION
+    The Get-VCFIdentityProviderIdByType cmdlet retrieves the ID of an identity provider by specifying "Embedded" or "Microsoft ADFS".
+    .EXAMPLE
+    Get-VCFIdentityProviderIdByType -type Embedded
+    This example shows how to get the ID of an embedded identity provider.
+    .EXAMPLE
+    Get-VCFIdentityProviderIdByType -type "Microsoft ADFS"
+    This example shows how to get the ID of an external identity provider.
+    #>
+    
+    param(
+    [Parameter(Mandatory = $true)][ValidateSet("Embedded","Microsoft ADFS")] [string]$type
+    )
+    Try {
+        $response = Get-VCFIdentityProviders
+        foreach ($item in $response.elements) {
+            if ($item.type -eq $type) {
+                return $item.id
+            }
+        }
+    } Catch {
+        ResponseException -Object $_
+    }
+}
+Export-ModuleMember -Function Get-VCFIdentityProviderIdByType
+
+function Remove-VCFIdentityProvider {
+    <#
+    .SYNOPSIS
+    Connects to the specified SDDC Manager & deletes an identity provider.
+    .DESCRIPTION
+    The Remove-VCFIdentityProvider cmdlet is used to delete an identity provider by specifying its type as "Embedded" or "Microsoft ADFS".
+    .EXAMPLE
+    Remove-VCFIdentityProvider -type Embedded -domainName sfo.rainpole.io.
+    This example shows how to delete an embedded identity provider with a specific domain name.
+    .EXAMPLE
+    Remove-VCFIdentityProvider -type "Microsoft ADFS".
+    This example shows how to delete an external identity provider.
+    #>
+
+    param(
+    [Parameter(Mandatory = $true)][ValidateSet("Embedded","Microsoft ADFS")] [string]$type,
+    [Parameter(Mandatory = $false)][ValidateNotNullOrEmpty()] [string]$domainName
+    )
+    Try {
+        if ($type -eq "Embedded") {
+            $id = (Get-VCFIdentityProviderIdByType -Type $type)
+            createHeader # Calls createHeader function to set Accept & Authorization
+            checkVCFToken # Calls the CheckVCFToken function to validate the access token and refresh if necessary
+            $uri = "https://$sddcManager/v1/identity-providers/$id/identity-sources/$domainName"
+            Invoke-RestMethod -Method DELETE -Uri $uri -Headers $headers # This API does not return a response
+        }
+        elseif ($type -eq "Microsoft ADFS") {
+            $id = (Get-VCFIdentityProviderIdByType -Type $type)
+            createHeader # Calls createHeader function to set Accept & Authorization
+            checkVCFToken # Calls the CheckVCFToken function to validate the access token and refresh if necessary
+            $uri = "https://$sddcManager/v1/identity-providers/$id"
+            Invoke-RestMethod -Method DELETE -Uri $uri -Headers $headers # This API does not return a response
+        }
+    } Catch {
+        ResponseException -Object $_
+    }
+}
+Export-ModuleMember -Function Remove-VCFIdentityProvider
+
+function New-VCFIdentityProvider {
```

Comment:
I think this function can be slightly simplified as well as adjust the formatting and adding the .PARAMETER to the `<# ... #>`

Take note of the indentation, casing, and use of `} else {`, `} elseif (`, `} Catch {` else on a single line.

Can you apply similar formatting and content on the rest of the cmdlets?

Also, let's include an example JSON specifications in `.\samples\iam\`

```powershell
Function New-VCFIdentityProvider {
    <#
        .SYNOPSIS
        Configures an identity provider.
        
        .DESCRIPTION
        The New-VCFIdentityProvider cmdlet configures an embedded or external identity provider
        from a JSON specification file.
        
        .EXAMPLE
        New-VCFIdentityProvider -type Embedded -json .\samples\iam\embeddedIdpSpec.json
        This example shows how to configure an embedded identity provider from the JSON specification file.
        
        .EXAMPLE
        New-VCFIdentityProvider -type "Microsoft ADFS" -json .\samples\iam\externalIdpSpec.json
        This example shows how to configure "Microsoft ADFS" identity provider from the JSON specification file.

        .PARAMETER type
        Specifies the type of the identity provider. One of: Embedded, Microsoft ADFS.

        .PARAMETER json
        Specifies the JSON specification file to be used.
    #>

    Param (
        [Parameter (Mandatory = $true)] [ValidateSet("Embedded","Microsoft ADFS")] [string]$type,
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [string]$json
    )

    Try {
        createHeader # Set the Accept and Authorization headers.
        checkVCFToken # Validate the access token and refresh, if necessary.
        if ($type -eq "Embedded") {
            $jsonBody = validateJsonInput -json $json
            $id = (Get-VCFIdentityProviderIdByType -Type Embedded)
            $uri = "https://$sddcManager/v1/identity-providers/$id/identity-sources"
        } elseif ($type -eq "Microsoft ADFS") {
            $jsonBody = validateJsonInput -json $json
            $uri = "https://$sddcManager/v1/identity-providers"
        }
        $response = Invoke-RestMethod -Method POST -Uri $uri -Headers $headers -ContentType application/json -Body $jsonBody
        $response
    } Catch {
        ResponseException -Object $_
    }
}
Export-ModuleMember -Function New-VCFIdentityProvider
````